db-engine/src/main/java/com/splicemachine/db/iapi/types/TypeId.java

```diff
@@ -1162,6 +1162,39 @@ public String getCorrespondingJavaTypeName(){
         return javaTypeName;
     }
 
+    /**
+     * Get the name of the corresponding Java Wrapper Class type.
+     * <p/>
+     * @return The name of the corresponding Java type.
+     */
+    public String getCorrespondingJavaWrapperTypeName(){
+        if(SanityManager.DEBUG){
+            SanityManager.ASSERT(javaTypeName!=null,
+                    "javaTypeName expected to be non-null");
+        }
+
+        switch (javaTypeName){
+            case "byte[]":
+                return "java.lang.Number[]";
+            case "short[]":
+                return "java.lang.Number[]";
+            case "long[]":
+                return "java.lang.Long[]";
+            case "char[]":
+                return "java.lang.Character[]";
+            case "boolean[]":
+                return "java.lang.Boolean[]";
+            case "double[]":
+                return "java.lang.Double[]";
+            case "float[]":
+                return "java.lang.Float[]";
+            case "int[]":
+                return "java.lang.Number[]";
```

Comment:
What is the reason for using abstract class java.lang.Number for some of these instead of the specific type, like java.lang.Integer?

Reply:
Array type of Objects among parameters are used for OUT parameters. Upon code generation, they would be converted to NumberDataType via the setValue method. For int, byte, short, it is the setValue((int) theValue) will be invoked.

Since there is no setValue method for type Integer, Byte, Short but only a method for Number, I convert them to Number.